include/internal/rcu.h

```diff
@@ -0,0 +1,32 @@
+/*
+ *
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OPENSSL_RCU_H
+# define OPENSSL_RCU_H
+# pragma once
+
+typedef void (*rcu_cb_fn)(void *data);
+
+typedef void* CRYPTO_RCU_LOCK;
```

Comment:
We can make this more type safe:

```c
typedef struct crypto_rcu_lock_st *CRYPTO_RCU_LOCK;
```

I am not sure why we don't do this for our existing mutex functions.

Reply:
fixed